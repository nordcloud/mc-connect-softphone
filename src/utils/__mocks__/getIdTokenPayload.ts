export const getIdTokenPayload = jest.fn().mockReturnValue({
  nonce: 'dummyIdTokenNonce',
});
