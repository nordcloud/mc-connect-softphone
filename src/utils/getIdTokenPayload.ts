import { IdTokenPayload } from '../types';

export function getIdTokenPayload(idToken: string) {
  const decodedToken = Buffer.from(idToken.split('.')[1], 'base64').toString();
  return JSON.parse(decodedToken) as IdTokenPayload;
}
