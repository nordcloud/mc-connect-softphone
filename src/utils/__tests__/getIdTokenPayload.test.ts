import { getIdTokenPayload } from '../getIdTokenPayload';

const idToken =
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c';

describe('getIdTokenPayload', () => {
  it('returns a correct value', () => {
    expect(getIdTokenPayload(idToken)).toMatchSnapshot();
  });
});
