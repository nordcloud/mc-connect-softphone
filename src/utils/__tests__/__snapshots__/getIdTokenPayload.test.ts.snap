// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIdTokenPayload returns a correct value 1`] = `
{
  "iat": 1516239022,
  "name": "John Doe",
  "sub": "1234567890",
}
`;
